### ISSUE TITLE
Allow repl launched from code to be handled from within notebook?

### ISSUE THREAD

**Issue by riffraff** [SciRuby/iruby#39]
The use case is the following: I have a long data massaging loop like

```
entries.each do 
  begin
    ... a lot of stuff that can go wrong
  rescue => e
   binding.pry
  end
```
if an error occur, I end up inside a `pry` session and I can properly inspect the data, error and eventually re-run the computation.

If I do this from within a notebook though the new `pry` session is launched in the underlying terminal, somewhat conflicting with the notebook logs.

It would be *awesome* to have the ability to launch a sub-repl from the notebook and interact with it.
Would this be possible, or it maybe already is, or does what I'm saying not even make sense? 

Anyway, thanks for your work on iRuby, it's awesome.

**Comment by v0dro** [comment]
If I understand correctly, you basically want a console inside your browser that can let you examine the contents of your objects right?

**Comment by riffraff** [comment]
yep

**Comment by minad** [comment]
Yes, this should be fixed. However the current support for pry is basically a hack if you take a look at the backend. Unfortunately the pry code is not really transparent. It would be best if someone of the pry guys could take a look into that and tell us how to interface iruby properly with pry. Maybe you could ask over there?
